File: chapter11_json.r
library(jsonlite)

data <- read_json( 'data.json' )

for( row in 1:length(data) ){

  print( paste( data[[ row ]]$id, data[[ row ]]$text ) )

  likes <- data[[ row ]]$likes

  for( user in likes ) {
    print( paste('    Liked by', user ) )
  }

}

## adding an entry

new_entry <- list(
  id = 3,
  text = 'This post has one Like',
  likes = c('John Smith')
)

## we make new entry a list to ensure it is percieved as a dictionary when joining items
data <- c( data, list( new_entry ) )

print( data )

write( toJSON( data, auto_unbox = TRUE ) , file = 'data2.json' )

File: chapter11_json.py
import json

data = json.load( open('data.json') )

for post in data:
  print( post['id'], post['text']  )

  liked_by = post['likes']

  for user in liked_by:
    print('    Liked by', user )

## adding an entry

new_entry = {
  'id' : 3,
  'text' : 'This post has one Like',
  'likes' : ['John Smith']
}

data.append( new_entry )

## storing new data

json.dump( data, open( 'data2.json', 'w' ) )
